src/sentry/api/serializers/models/incident.py

```diff
@@ -29,8 +37,33 @@ def get_attrs(self, item_list, user, **kwargs):
             results[incident] = {"projects": incident_projects.get(incident.id, [])}
             results[incident]["alert_rule"] = alert_rules.get(str(incident.alert_rule.id))
 
+            if "activities" in self.expand:
+                results[incident]["activities"] = serialize(
+                    list(
+                        IncidentActivity.objects.filter(incident=incident).select_related(
+                            "user", "incident"
+                        )
+                    )
+                )
+
+            if "seen_by" in self.expand:
+                (seen_by, has_seen) = self._get_incident_seen_list(incident, user)
+                results[incident]["seen_by"] = seen_by
+                results[incident]["has_seen"] = has_seen
+
         return results
 
+    def _get_incident_seen_list(self, incident, user):
+        seen_by_list = list(
+            User.objects.filter(incidentseen__incident=incident).order_by(
+                "-incidentseen__last_seen"
+            )
+        )
+
+        has_seen = any(seen_by for seen_by in seen_by_list if seen_by.id == user.id)
+
+        return serialize(seen_by_list), has_seen
```

Comment:
This logic will actually need to be modified to do a single query for all `incident` rows. Since previously this was in `DetailedIncidentSerializer`, which only every serialized one `Incident` at a time it was fine, but now that we will potentially be serializing many `Incident` rows it will be slow as is.